plugins/drivers/base/client.go

```diff
@@ -0,0 +1,80 @@
+package base
+
+import (
+	"errors"
+	"time"
+
+	"github.com/hashicorp/nomad/plugins/drivers/base/proto"
+	"golang.org/x/net/context"
+)
+
+type DriverClient interface {
```

Comment:
What is the point of this slimmed down interface?

Reply:
So this interface isn't fully built out yet. I need to finish adding other methods.

If I remember correctly I opted to separate the client and plugin side interfaces so you could pass in a context where appropriate on the client. For example on WaitTask. It feels like I started that code ages ago though and now think that might not be necessary?